### DISCUSSION TITLE
Alpine: no element provided to x-anchor

### DISCUSSION THREAD

**Discussion by Bi3Dev** [alpinejs/alpine#4004] [1. Help] [answered]
```
...
<div x-anchor.bottom-end.offset.3="document.getElementById($id('alpine-menu-button'))" x-menu:items x-cloak > ... </div>
...
```
The above code is throwing the "Alpine: no element provided to x-anchor" error and the dropdown menu is not showing when clicking the button. I tried adding the id='alpine-menu-button' to button and passing it as a id variable as well.

I am using Laravel 10 with breeze and livewire 3. I have removed the breeze import of alpine from app.js. I have also tried manually adding anchor package `npm install @alpinejs/anchor`.

Has anything changed since the screencast tutorial? I am not able to find the solution on my search so far and I am very new to livewire and alpine.

> [!NOTE]
> Rest of the code have been yoinked form the screencast tutorial the only difference being I have the following directory
>V-- buttonmenu
> |      |-- button.blade.php
> |      |-- item.blade.php
> |      |-- items.blade.php
> |      |-- close.blade.php
> |
> \>-- buttonmenu.blade.php

**Comment by ekwoka** [comment]
There isn't an element with that ID.

**Comment by calebporzio** [comment, marked as answer]
Make sure you have the Alpine UI plug-in included on the page. Of you're coming from the livewire screencasts, it's the script tag center link in the layout file. That should fix it 👍

**Reply by Bi3Dev** [reply]
I must have missed it because I was only looking at that specific video. Thank you very much that worked.

**Reply by alan9608** [reply]
Thanks Caleb.  I get so frustrated when something doesn't work.  Then you come along a solve the problem.  You are so worth the monthly stipend I contribute!
